### DISCUSSION TITLE
Plans for R5 Support

### DISCUSSION THREAD

**Discussion by Seqiiu** [FirelyTeam/firely-cql-sdk#660] [Ideas]
Hi,

I would like to ask if you have any plans to develop and support R5 in the firely-cql-sdk project? If so, when can we expect such a solution?

If you do not plan to support R5, could you advise on where to start to extend the CQL version to R5 on our own?

Thank you.

**Comment by alexzautke** [comment]
Hi @Seqiiu,

unfortunately we don't have any concrete plans to enable this within the next couple of month.
@baseTwo @ewoutkramer Could you point out where changes would be necessary to get this implemented?

**Reply by baseTwo** [reply]
@Seqiiu 

I have not looked at the changes in depth needed to support R5. As @alexzautke mentioned, this is not in the plan for 2025 Q1. Our main goal is to design the public API and then stabilize it. 

If you want to extend it yourself, start by changing the FhirRelease variable in cql-base.props from R4 to R5. You'd have to implement all the R5 changes yourself though, one example would be in CqlTypeToFhirTypeMapper adding support for CQL long to FHIR Integer64.
